syntax = "proto3";

package talko.registry;

// 定义服务实例对象
message ServiceInstance {
    bytes service_name = 1; // 服务名称
    bytes method_name  = 2; // 方法名称
    bytes address      = 3; // 服务节点的IP
    int32 port         = 4; // 服务节点的端口
}

// 定义请求类型
enum RequestType {
    REGISTER = 0; // 注册服务
    DISCOVER = 1; // 发现服务
}

// 定义服务请求
message ServiceRequest {
    RequestType     request_type = 1; // 请求类型
    ServiceInstance instance     = 2; // 服务实例对象
}

// 定义服务响应
message ServiceResponse {
    bool            success  = 1; // 发现是否成功
    bytes           err_msg  = 2; // 错误信息
    ServiceInstance instance = 3; // 服务实例对象
}